Islandora-CLAW/islandora_demo
# "Member Of" Gallery View

**GitHub Issue**:  related to Islandora-CLAW/CLAW#1032

* Makes use of Islandora-CLAW/islandora/pull/127
* Extended by https://github.com/seth-shaw-unlv/islandora_slick

# What does this Pull Request do?

Adds a new view with block using the new Islandora Image formatter on Media to show galleries of items using the field_member_of entity reference field. Provides blocks for displaying galleries of members of the current node and another block for displaying peers of the current node. It also updates the Content context reactions to include the peers block.

# What's new?
A in-depth description of the changes made by this PR. Technical details and
 possible side effects.

* Changes collection context to use the new members block.
* Changes repository content context to include the new peers block.
* Added Member Of View (with included blocks)
* Added islandora_demo_preprocess_views_view_unformatted(&$vars) which adds a "active-peer" class to a view row's div in the peer block if it matches the current node.
* Does this change require documentation to be updated? Probably. Not many significant changes, if so.
* Does this change add any new dependencies? No.
* Does this change require any other modifications to be made to the repository
 (ie. Regeneration activity, etc.)? No.
* Could this change impact execution of existing code? No.

# How should this be tested?

A description of what steps someone could take to:
* Make a collection and a few items that are members of that collection.
* View the collection to see the existing members list.
* View the items to see their existing display (only includes a link to the parent, not the peers)
* Apply the PR
* Import the feature
* View the collection, it will display a grid view of the members.
* View an item, it will now include a paged (assuming you have more than 5) block of peers in second sidebar.
* If you check out the source (or use Chrome's Dev Tools' "Inspect" feature) of an Item with peers, you will find the view result div for the current node will have an extra class of "active-peer" (i.e. `<div class="active-peer views-row">`). This allows a theme to "highlight" the current item in the results.

# Additional Notes:
I'm working on a [module to demonstrate using Slick Carousel](https://github.com/seth-shaw-unlv/islandora_slick) as an alternative.

Also note that the parent item does not need to be a collection. It can be any node using field_member_of, you will just need to place the blocks accordingly. Locally we are using a different content type that includes weight and sorting our view on that field.

# Interested parties
@Islandora-CLAW/committers
